test-utils/src/approx_eq.rs

```diff
@@ -112,21 +105,13 @@ macro_rules! assert_approx_eq {
     }};
 }
 
-/// Helper trait for the [`assert_approx_eq!`] macro.
-///
-/// Until we have GAT in rust this is meant to be implemented
-/// on a `&`-reference to the thing you want to implement it for.
```

Comment:
i made all changes in this module while i tried to figure out and fix the issue of the non-implemented trait, which turned out to be some kind of cyclic dependency issue. since we now have GATs, i think its still worth to keep the improvements, they could be split into a separate pr, if required.